### ISSUE TITLE
[iOS] Keyboard is Dismissed when trying to scroll the ListView

### ISSUE THREAD

**Issue by Soap-141** [unoplatform/uno#16180]
### Current behavior

Currently, on Android when we focus a field in a `ListView` footer, we can scroll it, but not on iOS.

Android
![Animation_142-ezgif com-optimize](https://github.com/unoplatform/uno/assets/57812285/7a8b1563-bcfa-493f-a26c-e3f480c068ab)

iOS
![Animation_143-ezgif com-optimize](https://github.com/unoplatform/uno/assets/57812285/585670f4-174c-4e94-adc7-ec6b58614281)

Haven't been able to test iOS on the provided sample application, but I was able to test it in UnoApplicationTemplate.

### Expected behavior

iOS behavior should match Android, we should be able to scroll the ListView while focusing on the textbox in the footer.

### How to reproduce it (as minimally and precisely as possible)

1. Download [UnoListViewKB.zip](https://github.com/unoplatform/uno/files/14906413/UnoListViewKB.zip)
2. Start it on iOS and Android
3. Notice the different behaviors

### Works on UWP/WinUI

None

### Environment

Uno.WinUI / Uno.WinUI.WebAssembly / Uno.WinUI.Skia

### NuGet package version(s)

5.3.0-dev.107

### Affected platforms

iOS

### IDE

Visual Studio 2022

### IDE version

17.9.5

### Anything else we need to know?

- This is also an issue on Uno 4 but on Android the whole footer is bring into view (not just the textbox).
- Related question: https://github.com/unoplatform/uno/discussions/16181

**Comment by Soap-141** [comment]
> @Soap-141 I am curious, are you able to reproduce the issue with current latest stable [5.1.104](https://www.nuget.org/packages/Uno.WinUI/5.1.104) as well ? Or just with latest dev 5.3.0-dev.107 ?

@agneszitte Yes pretty sure, it was the version used at first with the sample application before I updated to latest. And it's an issue with 4.10.37 or latest stable for Uno 4.

**Comment by agneszitte** [comment]
> > @Soap-141 I am curious, are you able to reproduce the issue with current latest stable [5.1.104](https://www.nuget.org/packages/Uno.WinUI/5.1.104) as well ? Or just with latest dev 5.3.0-dev.107 ?
> 
> @agneszitte Yes pretty sure, it was the version used at first with the sample application before I updated to latest. And it's an issue with 4.10.37 or latest stable for Uno 4.

Thanks a lot for all the version details @Soap-141, really appreciated. We will look into the issue.

**Comment by agneszitte** [comment]
@MartinZikmund, @dr1rrb, @ajpinedam 
I would like your input for this issue please
As what would be the correct default behavior here (based on Windows I imagine)?
Will it involve something like [scrolldismisseskeyboard](https://developer.apple.com/documentation/swiftui/view/scrolldismisseskeyboard(_:)) for iOS ?

**Comment by Youssef1313** [comment]
This feels like a focus bug where focus is lost when it shouldn't?

**Comment by MartinZikmund** [comment]
This is only happening on native
